developer_tools/get_coverage_tests.py

```diff
@@ -11,97 +11,177 @@
+          startReading = True
+          collectSpecs = False
+        else: # This line is the start of a differ
+          startReading = False
+          collectSpecs = False
+        depth += 1 # Going down a level
+
       if startReading:
-        splitted = line.strip().split('=')
+        splitted = line.strip().split('=', 1)
         if len(splitted) == 2:
           testSpecs[splitted[0].strip()] = splitted[1].replace("'", "").replace('"', '').strip()
-      if line.strip().startswith("[./"):
-        startReading = True
-        collectSpecs = False
+
       if collectSpecs:
-        # collect specs
-        testFileList.append(testSpecs)
+        # Collect specs
+        testList.append(testSpecs)
         collectSpecs = False
         testSpecs = {}
-    if root not in doTests.keys():
+    # Now we have all the specs collected
+
+    if groupBy == 'directory' and root not in doTests.keys():
       doTests[root] = []
-    # now we have all the specs collected
-    for spec in testFileList:
-      # check if test is skipped or an executable is required
+
+    for spec in testList:
+      # Check if test is skipped or an executable is required
       if "required_executable" in spec or "skip" in spec:
         continue
       if "input" not in spec:
         continue
       testType = spec.get('type', "notfound").strip()
-      newTest = spec['input'].split()[0]
-      testInterfaceOnly = False
-      if 'test_interface_only' in spec:
-        testInterfaceOnly = True if spec['test_interface_only'].lower() == 'true' else False
-      if whichTests in [1, 3]:
-        if newTest.endswith(suffix) and testType.lower() not in 'ravenpython':
+      newTestFile = spec['input'].split()[0]
+
+      if whichTests == 4:
+        addTest = True
+      else:
+        addTest = False
+        testInterfaceOnly = False
+        if 'test_interface_only' in spec:
+          testInterfaceOnly = True if spec['test_interface_only'].lower() == 'true' else False
+        if newTestFile.endswith('.xml'):
           if whichTests == 3 and testInterfaceOnly:
-            doTests[root].append(newTest)
+            addTest = True
           elif whichTests == 1 and not testInterfaceOnly:
-            doTests[root].append(newTest)
-      else:
-        if newTest.endswith(suffix) and testType.lower() in 'ravenpython':
-          doTests[root].append(newTest)
+            addTest = True
+        elif newTestFile.endswith('.py'):
+          if whichTests == 2 and testType.lower() != 'crowpython':
+            addTest = True
+
+      if addTest:
+        if groupBy == 'directory':
+          doTests[root].append(newTestFile)
+        else:
+          doTests[spec['test_name']] = spec
+
   return doTests
 
 if __name__ == '__main__':
-  # skip the expected failed tests
-  skipFails = True if '--skip-fails' in sys.argv else  False
-  if '--get-python-tests' in sys.argv:
-    # unit tests flag has priority over interface check
+  if args.find_test_names:
```

Comment:
very minor, but these variables (e.g. `find_test_names` should be in cameBack case). I believe the strings should be fine.